adapters/bluesea/bluesea.go

```diff
@@ -0,0 +1,158 @@
+package bluesea
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type blueseaBidExt struct {
+	MediaType string `json:"mediatype"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	impCount := len(request.Imp)
+
+	if impCount == 0 {
+		err := &errortypes.BadInput{
+			Message: "Empty Imp objects",
+		}
+		return nil, []error{err}
+	}
+
+	requestDatas := make([]*adapters.RequestData, 0, impCount)
+	errs := make([]error, 0, impCount)
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	for _, imp := range request.Imp {
+		blueseaImpExt, err := extraImpExt(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		reqJson, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		queryParams := url.Values{}
+		queryParams.Add("pubid", blueseaImpExt.PubId)
+		queryParams.Add("token", blueseaImpExt.Token)
+		queryString := queryParams.Encode()
+		requestData := &adapters.RequestData{
+			Method:  "POST",
+			Uri:     fmt.Sprintf("%s?%s", a.endpoint, queryString),
+			Body:    reqJson,
+			Headers: headers,
+		}
+		requestDatas = append(requestDatas, requestData)
+	}
+	return requestDatas, errs
```

Comment:
lets add length check and error out in case `requestDatas` is empty after looping on `request.Imp `

```
for  _, imp := range request.Imp {
 ...
}

if len(requestDatas) == 0 {
     errs = append(errs, fmt.Errorf("requestDatas is empty"))
}

return requestDatas, errs

```